core/perf_test/PerfTestBlasKernels.hpp

```diff
@@ -74,7 +74,9 @@ struct Dot {
   KOKKOS_INLINE_FUNCTION
   static void join(volatile value_type& update,
                    const volatile value_type& source) {
-    update += source;
+    // compound assignment to object of volatile-qualified type is deprecated in
+    // c++20
+    update = update + source;
```

Comment:
Given that these are not atomic accesses, do we think this is doing the right thing?  That is the reason which C++20 deprecated many uses of volatile for in the first place.

Reply:
That's a fair question...
@crtrott is there any reason to worry this is not doing the right thing as @nliber mentioned?

Reply:
I think its fine. Our use of volatile is strictly for "don't optimize this access away" never to deal with race conditions